modules/mbedtls/configs/config-tls-generic.h

```diff
@@ -458,6 +458,14 @@
 #if defined(CONFIG_MBEDTLS_PSA_CRYPTO_C)
 #define MBEDTLS_PSA_CRYPTO_C
 #define MBEDTLS_USE_PSA_CRYPTO
+
+#if defined(CONFIG_BOARD_NRF52_BSIM) || defined(CONFIG_BOARD_NATIVE_POSIX)
```

Comment:
@joerchan ble mesh requires these options to be able to use keys with persistent lifetime `PSA_KEY_LIFETIME_PERSISTENT`.

Reply:
Selecting these options based on board seems very strange to me.
Can you explain why these boards need the configurations?

Reply:
@alxelax Is there a kconfig for the ITS based on POSIX file API? It seems better to enable these options based on this feature instead of the boards.

Reply:
@joerchan, there are no configs for ITS. Right now whatever you build is not able to be built if you chose PSA and POSIX incompatible platforms. I doubt anything uses mbedTLS PSA that is run in Zephyr's CI. There are bugs that I fix in this PR that does not allow building mbedTLS PSA with Zephyr's configs at all. Probably only specific configurations.
I'm pretty sure we (as ble mesh maintainers) will have to create a specific mesh configuration file to configure mbedtTLS properly but for now default config is OK with this addition.
mbedTLS ITS doesn't depend on features. It lacks implementations for platforms. It is why I added dependency on platform.